.travis.yml

```diff
@@ -22,33 +26,64 @@ matrix:
     - os: linux
       language: python
       python: "2.7"
+      env:
+        TEST_COMPILER=true
     - os: linux
       language: python
       python: "3.5"
       env:
+        TEST_COMPILER=true
         TEST_REGISTRY=true
       services:
         - postgresql
     - os: linux
       language: python
       python: "3.6"
       env:
+        TEST_COMPILER=true
         TEST_REGISTRY=true
       services:
         - postgresql
+    - os: linux
+      language: node_js
+      cache:
+        directories:
+          - $HOME/.npm
+      node_js:
+        - node
+        - lts
+      env:
+        TEST_CATALOG=true
 
 install:
-  # install quilt from setup.py
-  - travis_retry pip install ./compiler['tests']
+  - |
+    # install quilt from setup.py
+    if [[ "$TEST_COMPILER" = true ]]; then
+      travis_retry pip install ./compiler['tests']
+    fi
   - |
     if [[ "$TEST_REGISTRY" = true ]]; then
       travis_retry pip install -r registry/requirements.txt
       pip install registry
     fi
+  - |
+    if [[ "$TEST_CATALOG" = true ]]; then
+      cd catalog && npm ci
```

Comment:
ok, i'll adjust this

Reply:
i've just realised that python and js stuff is never run in single context because of conditionals (e.g. this one: `if [[ "$TEST_CATALOG" = true ]]; then`), so i'ts nothing to worry about